drivers/counter/counter_mchp_xec.c

```diff
@@ -243,14 +243,19 @@ static int counter_xec_set_top_value(const struct device *dev,
 
 static void counter_xec_isr(const struct device *dev)
 {
-	BTMR_Type *counter = COUNTER_XEC_REG_BASE(dev);
+	struct btmr_regs *counter = COUNTER_XEC_REG_BASE(dev);
 	const struct counter_xec_config *counter_cfg = COUNTER_XEC_CONFIG(dev);
 	struct counter_xec_data *data = COUNTER_XEC_DATA(dev);
 	counter_alarm_callback_t alarm_cb;
 	void *user_data;
 
 	counter->STS = MCHP_BTMR_STS_ACTIVE;
+
+#if defined(CONFIG_BOARD_MEC172XEVB_ASSY6906)
```

Comment:
Having board specific ifdef in driver code should be avoided.

Reply:
@galak MEC152 uses a HAL and MEC172x does not the header definitions are very different.
Any suggestions @jvasanth1

Reply:
Is this really board specific or SoC specific?

Reply:
It's SoC specific maybe the change could be changed to CONFIG_SOC_SERIES_MEC172X

Reply:
This is changed in the latest version